agent/metrics_test.go

```diff
@@ -203,6 +225,32 @@ func TestAgent_OneTwelveRPCMetrics(t *testing.T) {
 		// make sure we see 3 Status.Ping metrics corresponding to the calls we made above
 		assertLabelWithValueForMetricExistsNTime(t, respRec, metricsPrefix+"_rpc_server_call", "method", "Status.Ping", 3)
 	})
+
+	t.Run("RPC calls with elapsed time below 1ms are reported as decimal", func(t *testing.T) {
+		metricsPrefix := "new_rpc_metrics_3"
+		allowRPCMetricRule := metricsPrefix + "." + strings.Join(middleware.OneTwelveRPCSummary[0].Name, ".")
+		hcl := fmt.Sprintf(`
+		telemetry = {
+			prometheus_retention_time = "5s"
+			disable_hostname = true
+			metrics_prefix = "%s"
+			prefix_filter = ["+%s"]
+		}
+		`, metricsPrefix, allowRPCMetricRule)
+
+		a := StartTestAgent(t, TestAgent{HCL: hcl})
+		defer a.Shutdown()
+
+		var out struct{}
+		err := a.RPC("Status.Ping", struct{}{}, &out)
+		require.NoError(t, err)
+
+		respRec := httptest.NewRecorder()
+		recordPromMetrics(t, a, respRec)
+
+		assertMetricsWithLabelIsNonZero(t, respRec, "method", "Status.Ping")
```

Comment:
Awesome, thanks for adding this in!

We don't actually need a full `t.Run` to use this assertion. Let's move this line (251) right after the 226 line and remove this test since this one and the above are essentially testing the same use case.